### PR TITLE
Lazy import of bitsandbytes

### PR DESCRIPTION
**Pull request by BenjaminBossan**
Should fix various issues reported here and elsewhere, e.g. #559, https://github.com/huggingface/accelerate/issues/2216.

## Description

Previously, we imported from bitsandbytes eagerly if the package was installed. This caused two major issues:

- Slow loading time of PEFT (~4 sec) when bnb is installed
- Errors with multiprocessing because bnb initializes CUDA -- users had to load PEFT lazily or else would get:

 > RuntimeError: Cannot re-initialize CUDA in forked subprocess. To use CUDA with multiprocessing, you must use the 'spawn' start method

This commit fixes both issues by importing bitsandbytes lazily. PEFT import time is now reduced to ~2sec.

## Notes

Implementation-wise, I use a combination of local imports and module-level `__getattr__`. The latter was introduced in Python 3.7 and should therefore be safe to use.

As for testing, it is not as straightforward as just adding a new unit test. The issue is that when multiple tests are run, depending on the order, bitsandbytes may already be imported once the new test runs. To avoid this issue, I created a completely separate testing script that is run in CI. It is executed on the nightly multi-GPU CI run. Locally, it passes, :crossed_fingers: that it also passes there.

If anyone has a better idea, please let me know.

### CODE REVIEW

**Review by pacman100 — APPROVED** [review]
Thank you @BenjaminBossan for fixes the issues as well as making the import faster, LGTM! 🚀

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
           source activate peft
           pip install -e . --no-deps
           pip install pytest-reportlog
-      
+
```
Comment (younesbelkada) [inline .github/workflows/nightly.yml:37]: Why this diff is here?

Comment (BenjaminBossan) [reply .github/workflows/nightly.yml]: Note that I moved the test to the proper unit tests using a trick that Zach showed, so I undid all the changes here (i.e. trailing whitespaces are back).

**Review by muellerzr — APPROVED** [review]
Thanks for fixing this! Great job :)

**Pull request merged**